src/tools/haxelib/Main.hx

```diff
@@ -1341,42 +1341,51 @@ class Main {
 		print("Library "+prj+" current version is now "+version);
 	}
 
-    function checkRec( prj : String, version : String, l : List<{ rep : String, project : String, version : String, info : Infos }> ) {
-        var reps = getRepositories();
-        var found = false;
-        for (rep in reps) {
-		    var pdir = rep + Data.safe(prj);
-                if( !FileSystem.exists(pdir) ) {
-                    continue;
-                }
-                found = true;
-		var version = if( version != null ) version else getCurrent(pdir);
-		var vdir = pdir + "/" + Data.safe(version);
-		if( vdir.endsWith("dev") )
-			vdir = getDev(pdir);
-		if( !FileSystem.exists(vdir) )
-			throw "Library "+prj+" version "+version+" is not installed";
-		for( p in l ) {
-			if( p.project == prj ) {
-				if( p.version == version )
-					return;
-				throw "Library "+prj+" has two version included "+version+" and "+p.version;
+	function checkRec( prj : String, requestedVersion : String, l : List<{ rep : String, project : String, version : String, info : Infos }> ) {
+		var reps = getRepositories();
+		var found = false;
+		for (rep in reps) {
+			var pdir = rep + Data.safe(prj);
+			if( !FileSystem.exists(pdir) ) {
+				continue;
 			}
+			var version;
+			try {
+				version = requestedVersion != null ? requestedVersion : getCurrent(pdir);
+			} catch(e:Dynamic) {
+				print("No .current file for haxelib dir " + rep + " for haxelib " + prj);
+				continue;
+			};
+
+
+			var vdir = pdir + "/" + Data.safe(version);
+			if( vdir.endsWith("dev") )
+				vdir = getDev(pdir);
+			if( !FileSystem.exists(vdir) ) {
```

Comment:
Same problem here, if explicit version was requested it would error out without checking all the paths.